tests/integration_tests/datasource_tests.py

```diff
@@ -168,25 +158,21 @@ def test_save(self):
             else:
                 self.assertEqual(resp[k], datasource_post[k])
 
-    def save_datasource_from_dict(self, datasource_dict):
```

Comment:
Hmmm. The local variable was never used but instead the global variable was. Seems questionable.